ixmp/core.py

```diff
@@ -20,30 +23,49 @@
 IAMC_IDX = ['model', 'scenario', 'region', 'variable', 'unit']
 
 
-# %% Java Virtual Machine start-up
-
 def start_jvm(jvmargs=None):
+    """Start the Java Virtual Machine via JPype.
+
+    Parameters
+    ----------
+    jvmargs : str or list of str, optional
+        Additional arguments to pass to :meth:`jpype.startJVM`.
+    """
+    # TODO change the jvmargs default to [] instead of None
     if jpype.isJVMStarted():
         return
 
-    module_root = os.path.dirname(__file__)
-    jarfile = os.path.join(module_root, 'ixmp.jar')
-    module_lib = os.path.join(module_root, 'lib')
-    module_jars = [os.path.join(module_lib, f) for f in os.listdir(module_lib)]
+    jvmargs = jvmargs or []
+
+    # Arguments
+    args = [jpype.getDefaultJVMPath()]
+
+    # Add the ixmp root directory, ixmp.jar and bundled .jar and .dll files to
+    # the classpath
+    module_root = Path(__file__).parent
+    jarfile = module_root / 'ixmp.jar'
+    module_jars = list(module_root.glob('lib/*'))
+    classpath = map(str, [module_root, jarfile] + list(module_jars))
+
     sep = ';' if os.name == 'nt' else ':'
-    classpath = sep.join([module_root, jarfile] + module_jars)
-    args = ["-Djava.class.path={}".format(classpath)]
-    if jvmargs is not None:
-        args += jvmargs if isinstance(jvmargs, list) else [jvmargs]
-    jpype.startJVM(jpype.getDefaultJVMPath(), *args)
+    args.append('-Djava.class.path={}'.format(sep.join(classpath)))
+
+    # Add user args
+    args.extend(jvmargs if isinstance(jvmargs, list) else [jvmargs])
+
+    # Explicitly set the convertStrings option for JPype 0.7
+    # TODO set to False, fix code, and deprecate for JPype 0.8
```

Comment:
@khaeru do you have an idea why setting convertString to false is recommended/default option in jpype >= 0.8?

As far as I see from sources having this option as true will force always converting Java strings to Python strings (see [function  declaration called](https://github.com/jpype-project/jpype/blob/49faa31c26a10377ebdeb4fa6a89d5ba9e8f97f5/native/python/include/jp_pythontypes.h#L311)). I can imaging that it might not always necessary to do conversion because of performance reasons, but requiring it to be called explicitly every time (if that's the other option) doesn't look to me best idea.

Reply:
No, I have no idea.